jscomp/reason_outcome_printer/tweaked_reason_oprint.ml

```diff
@@ -251,6 +251,71 @@ and print_simple_out_type ppf =
     Otyp_class (ng, id, tyl) ->
       fprintf ppf "@[%s#%a%a@]" (if ng then "_" else "")
         print_ident id print_typargs tyl
+#if undefined BS_NO_COMPILER_PATCH then         
+  | Otyp_constr ( (Oide_dot ((Oide_dot (Oide_ident "Js", "Internal")),
+                             ("fn" | "meth" as name )) as id) ,
+                 ([Otyp_variant(_,Ovar_fields [ variant, _, tys], _,_); result] as tyl))
+    ->
+      (* Otyp_arrow*)
+      let make tys result =
+        if tys = [] then
```

Comment:
Since we're pattern-matching against `tys` a few lines down, maybe it makes to move this case into the pattern match itself?